provider.go

```diff
@@ -77,6 +83,63 @@ func readJWTToken(path string) (string, error) {
 	return string(bytes.TrimSpace(data)), nil
 }
 
+func splitSecretPath(secretPath string) (string, string) {
+	s := regexp.MustCompile("/+").Split(secretPath, 3)
+	return s[1], s[2]
+}
+
+func (p *Provider) getMountInfo(mountName string, token string) (string, string, error) {
+	client, err := p.createHTTPClient()
+	if err != nil {
+		return "", "", err
+	}
+
+	addr := p.VaultAddress + "/v1/sys/mounts"
+	req, err := http.NewRequest(http.MethodGet, addr, nil)
+	// Set vault token.
+	req.Header.Set("X-Vault-Token", token)
```

Comment:
Also set `X-Vault-Request` to `"true"`. This is a [recently-added SSRF mitigation](https://github.com/hashicorp/vault/blob/14c7398841a1b0794a42882f7a6d223d6cbecf59/vendor/github.com/hashicorp/vault/api/client.go#L438).